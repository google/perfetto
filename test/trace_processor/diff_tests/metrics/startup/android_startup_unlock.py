#!/usr/bin/env python3
# Copyright (C) 2018 The Android Open Source Project
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from os import sys

import synth_common

trace = synth_common.create_trace()
trace.add_packet()
trace.add_process(1, 0, 'init')
trace.add_process(2, 1, 'system_server')
trace.add_process(3, 1, 'com.google.android.calendar', 10003)
trace.add_process(4, 1, 'com.android.systemui', 10004)

trace.add_package_list(
    ts=1, name='com.google.android.calendar', uid=10003, version_code=123)
trace.add_package_list(
    ts=1, name='com.android.systemui', uid=10004, version_code=123)

trace.add_ftrace_packet(cpu=0)

trace.add_atrace_async_begin(ts=100, tid=2, pid=2, buf='launchingActivity#1')
trace.add_atrace_async_end(ts=200, tid=2, pid=2, buf='launchingActivity#1')

trace.add_atrace_begin(
    ts=130, tid=2, pid=4, buf='KeyguardUpdateMonitor#onAuthenticationSucceeded')
trace.add_atrace_end(ts=133, tid=2, pid=4)

trace.add_atrace_instant(
    ts=201,
    tid=2,
    pid=2,
    buf='launchingActivity#1:completed:com.google.android.calendar')

sys.stdout.buffer.write(trace.trace.SerializeToString())
