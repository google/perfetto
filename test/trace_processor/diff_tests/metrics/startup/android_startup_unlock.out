android_startup {
  startup {
    startup_id: 1
    cpu_count: 0
    package_name: "com.google.android.calendar"
    zygote_new_process: false
    to_first_frame {
      dur_ns: 100
      main_thread_by_task_state {
        running_dur_ns: 0
        runnable_dur_ns: 0
        uninterruptible_sleep_dur_ns: 0
        interruptible_sleep_dur_ns: 0
        uninterruptible_io_sleep_dur_ns: 0
        uninterruptible_non_io_sleep_dur_ns: 0
      }
      other_processes_spawned_count: 0
      dur_ms: 0.0001
    }
    activity_hosting_process_count: 0
    event_timestamps {
      intent_received: 100
      first_frame: 200
    }
    system_state {
      dex2oat_running: false
      installd_running: false
      broadcast_dispatched_count: 0
      broadcast_received_count: 0
      installd_dur_ns: 0
      dex2oat_dur_ns: 0
    }
    slow_start_reason: "Unlock running during launch"
    slow_start_reason_with_details {
      reason_id: UNLOCK_RUNNING
      reason: "Unlock running during launch"
      severity: ERROR
      expected_value {
        value: 0
        unit: TRUE_OR_FALSE
      }
      actual_value {
        value: 1
      }
      launch_dur: 100
      trace_slice_sections {
        slice_section {
          start_timestamp: 130
          end_timestamp: 133
          slice_id: 1
          slice_name: "KeyguardUpdateMonitor#onAuthenticationSucceeded"
          thread_tid: 2
        }
        start_timestamp: 130
        end_timestamp: 133
      }
    }
  }
}
